### PR TITLE
[Android] prepare native libs for c-api @open sesame 11/10 13:27

### PR DESCRIPTION
**Pull request by jaeyun-jung**
update build script to provide native libraries using nnstreamer c-api.

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

**Comment by myungjoo** [comment]
After merging this, please provide a document (somewhere appropriate) on using C-API in Android, especially, please try to save people from not initializing appropriately.

File: api/android/api/src/main/jni/nnstreamer-native-api.c [hunk 1]
```diff
@@ -506,6 +505,23 @@ Java_org_nnsuite_nnstreamer_NNStreamer_nativeInitialize (JNIEnv * env, jclass cl
...
+ * @brief Native method to initialize NNStreamer.
+ */
+jboolean
+Java_org_nnsuite_nnstreamer_NNStreamer_nativeInitialize (JNIEnv * env, jclass clazz,
+    jobject context)
+{
+  nns_logi ("Called native initialize.");
+
+  if (!gst_is_initialized ()) {
+    nns_loge ("GStreamer is not initialized.");
```
Comment (myungjoo) [inline api/android/api/src/main/jni/nnstreamer-native-api.c]: This gotta be FATAL, not ERROR :)

File: api/android/api/src/main/jni/nnstreamer-native-api.c [hunk 2]
```diff
@@ -506,6 +505,23 @@ Java_org_nnsuite_nnstreamer_NNStreamer_nativeInitialize (JNIEnv * env, jclass cl
...
+    return JNI_FALSE;
+  }
+
+  return nnstreamer_native_initialize ();
```
Comment (myungjoo) [inline api/android/api/src/main/jni/nnstreamer-native-api.c]: calling ```gst_init_check``` after calling ```gst_is_initialized```?

Comment (jaeyun-jung) [reply api/android/api/src/main/jni/nnstreamer-native-api.c]: I will move gst_is_initialized() into nnstreamer_native_initialize(), and remove ml_initialize_gstreamer().
(GStreamer.java should be called and ml_initialize_gstreamer() is unnecessary.)

Comment (jaeyun-jung) [reply api/android/api/src/main/jni/nnstreamer-native-api.c]: Updated

File: api/android/build-android-lib.sh [hunk 3]
```diff
@@ -180,13 +180,35 @@ if [[ -e $nnstreamer_android_api_lib ]]; then
     mkdir -p $result_directory
     cp $nnstreamer_android_api_lib $result_directory/$nnstreamer_lib_name-$today.aar
 
+    # Prepare native libraries and header files for C-API
```
Comment (leemgs) [inline api/android/build-android-lib.sh]: Looks good.

**Pull request merged**